txtdirect.go

```diff
@@ -73,31 +73,37 @@ func contains(array []string, word string) bool {
 // and if it's not provided it will check txtdirect config for
 // default fallback address
 func fallback(w http.ResponseWriter, r *http.Request, fallback, recordType, fallbackType string, code int, c Config) {
-	if fallback != "" {
-		log.Printf("[txtdirect]: %s > %s", r.Host+r.URL.Path, fallback)
-		if code == http.StatusMovedPermanently {
-			w.Header().Add("Cache-Control", fmt.Sprintf("max-age=%d", status301CacheAge))
-		}
-		w.Header().Add("Status-Code", strconv.Itoa(code))
+	if code == http.StatusMovedPermanently {
+		w.Header().Add("Cache-Control", fmt.Sprintf("max-age=%d", status301CacheAge))
+	}
+	w.Header().Add("Status-Code", strconv.Itoa(code))
+
+	if fallback != "" && fallbackType != "global" {
 		http.Redirect(w, r, fallback, code)
 		if c.Prometheus.Enable {
 			FallbacksCount.WithLabelValues(r.Host, recordType, fallbackType).Add(1)
 			RequestsByStatus.WithLabelValues(r.URL.Host, strconv.Itoa(code)).Add(1)
 		}
+	} else if contains(c.Enable, "www") {
+		s := strings.Join([]string{defaultProtocol, "://", defaultSub, ".", r.URL.Host}, "")
+		if c.Prometheus.Enable {
+			RequestsByStatus.WithLabelValues(r.URL.Host, strconv.Itoa(code)).Add(1)
+		}
+		http.Redirect(w, r, s, code)
 	} else if c.Redirect != "" {
-		if contains(c.Enable, "www") {
-			log.Printf("[txtdirect]: %s > %s", r.Host+r.URL.Path, c.Redirect)
-			w.Header().Set("Content-Type", "")
-			w.Header().Add("Status-Code", strconv.Itoa(http.StatusForbidden))
-			http.Redirect(w, r, c.Redirect, http.StatusForbidden)
-			if c.Prometheus.Enable {
-				FallbacksCount.WithLabelValues(r.Host, recordType, "global").Add(1)
-				RequestsByStatus.WithLabelValues(r.URL.Host, string(http.StatusForbidden)).Add(1)
-			}
+		w.Header().Set("Content-Type", "")
+		w.Header().Set("Status-Code", strconv.Itoa(http.StatusForbidden))
```

Comment:
Why do we us StatusForbidden for the global fallback?
Also why add the specific content-type? Shouldn't this me a normal redirect using the global "redirect" config?

Reply:
Can't remember the reason we had for using StatusForbidden. It doesn't make sense to me too. What do you think it should be? 302 or the given code to the function? (Think the only place that we don't give a status code to fallback is after checking if r.host is an IP)

Also not sure about the Content-Type but we had this comment in our previous code:
`Empty Content-Type to prevent http.Redirect from writing an html response body`
But think it's actually opposite of the code that is inside net/http package.

https://github.com/golang/go/blob/master/src/net/http/server.go#L2090-L2096

Reply:
Left over copy and paste code from the blacklist maybe?